ffi/blitbuffer.lua

```diff
@@ -1818,27 +1792,21 @@ ffi.metatype("ColorRGB16", ColorRGB16_mt)
 ffi.metatype("ColorRGB24", ColorRGB24_mt)
 ffi.metatype("ColorRGB32", ColorRGB32_mt)
 
-function BB.new(width, height, buffertype, dataptr, pitch, phys_width, phys_height)
+function BB.new(width, height, buffertype, dataptr, pitch, pixel_pitch)
     local bb = nil
     buffertype = buffertype or TYPE_BB8
-    -- Remember the fb's _virtual dimensions if we specified them, as we'll need 'em for fast blitting codepaths
-    phys_width = phys_width or width
-    phys_height = phys_height or height
-    if pitch == nil then
-        if buffertype == TYPE_BB4 then pitch = band(1, width) + rshift(width, 1)
-        elseif buffertype == TYPE_BB8 then pitch = width
-        elseif buffertype == TYPE_BB8A then pitch = lshift(width, 1)
-        elseif buffertype == TYPE_BBRGB16 then pitch = lshift(width, 1)
-        elseif buffertype == TYPE_BBRGB24 then pitch = width * 3
-        elseif buffertype == TYPE_BBRGB32 then pitch = lshift(width, 2)
-        end
-    end
-    if buffertype == TYPE_BB4 then bb = BlitBuffer4(width, phys_width, height, phys_height, pitch, nil, 0)
-    elseif buffertype == TYPE_BB8 then bb = BlitBuffer8(width, phys_width, height, phys_height, pitch, nil, 0)
-    elseif buffertype == TYPE_BB8A then bb = BlitBuffer8A(width, phys_width, height, phys_height, pitch, nil, 0)
-    elseif buffertype == TYPE_BBRGB16 then bb = BlitBufferRGB16(width, phys_width, height, phys_height, pitch, nil, 0)
-    elseif buffertype == TYPE_BBRGB24 then bb = BlitBufferRGB24(width, phys_width, height, phys_height, pitch, nil, 0)
-    elseif buffertype == TYPE_BBRGB32 then bb = BlitBufferRGB32(width, phys_width, height, phys_height, pitch, nil, 0)
+    local pixel_size = assert(BB.TYPE_TO_BPP[tonumber(buffertype)], "unknown buffer type " .. tostring(buffertype))
+
+    -- If no pitch is given, make up one from buffer width
+    pitch = pitch or rshift(width * pixel_size + 7, 3)
+    pixel_pitch = pixel_pitch or (pitch / pixel_size)
```

Comment:
It's been a long time since I looked at this module, and I'm not competent enough to give any comment.
The word "pitch" was already there, but I'm not familiar with the term and its meaning here - and that rshift formula looks a bit cryptic to me :)
Can you add a comment somewhere to define what pitch/pixel_pitch actually is (the other stuff, bpp, something_w, I can guess, pitch I can't :/)

Reply:
I'm not necessarily super fond of the (previous) usage of pitch here, but the alternative is something like stride, which isn't any better? :D.

We *could* arguably come up with something ourselves line `(scan)line_length` (in pixels) / `(scan)line_size` (in bytes) or something?

---

>> 3 is / 8

Reply:
Barring something custom, `stride` is possibly more commonly recognizable in that context than `pitch`?

In which case `pixel_stride` would work for me for its pixel counterpart.

Reply:
Pitch is kind of a misnomer indeed. I'm tempted to wholesale replace pitch for stride everywhere (not just comments).

> >> 3 is / 8

Hmm, this code looks outright wrong, probably should be

```
pixel_pitch = pixel_pitch or (pitch * 8 / pixel_size)
```

Reply:
also probably rename pixel_size to bits_per_pixel to make things more obvious

Reply:
> Barring something custom, stride is possibly more commonly recognizable in that context than pitch?

Not for me :/ you haven't yet told me clearly what it is :)
```
stride (n): a long step in walking.
stride (v): to pass over or across in one long step: 
```
So, I guess it is the length of a bb _horizontal_ (?) line in... bytes ? So, it's width * bpp/8 ? Right ?

So, you're talking about scan_line_size / scan_line_nbpixels ?
Please just explain somewhere what is this thing, whatever  you decide to name it :) So I (and future readers) don't have to guess and stay unsure of my (their) guess.

Reply:
The physical analogy is because it's the amount of bytes you have to walk over to step across a scanline.

Which, in plaintext, translates to the length of a scanline (i.e., visible + padding) in bytes ;).

Reply:
@poire-z Stride is specifically for bitmap blitting. It's either amount of bytes or pixels until next row (assuming no rotation), usually both are tracked so as to avoid issues with bpp shifts. Pitch is conceptually related to stride, but it can measured in any unit - for instance, font libraries often use *points*. Pitch is typically used for *on screen* transforms (eg path tracing of font raster with subpixel antialiasing), whereas stride is exclusively about computing next row, pixel or byte.

Reply:
Thanks !
So, I guess a better comment on these 2 lines would be enough (for me to get it):
```
52    int pixel_pitch;   // nb of pixels from the start of a line to the start of next line
53    int h;
54    int pitch;         // nb of bytes from the start of a line to the start of next line
```